File: ant-sdk/antd-py/examples/03_chunks.py
"""Example 03: Store and retrieve raw chunks.

Chunks are the lowest-level storage primitive on Autonomi.
"""

from antd import AntdClient

client = AntdClient()

# Store a raw chunk
raw_data = b"Raw chunk content for direct storage"
result = client.chunk_put(raw_data)
print(f"Chunk stored at: {result.address}")
print(f"Cost: {result.cost} atto tokens")

# Retrieve the chunk
retrieved = client.chunk_get(result.address)
print(f"Retrieved {len(retrieved)} bytes")

assert retrieved == raw_data, "Chunk round-trip mismatch!"
print("Chunk round-trip OK!")

File: ant-sdk/antd-js/examples/03-chunks.ts
/**
 * Example 03: Store and retrieve raw chunks.
 *
 * Chunks are the lowest-level storage primitive on Autonomi.
 */

import { createClient } from "../src/index.js";

const client = createClient();

// Store a raw chunk
const rawData = Buffer.from("Raw chunk content for direct storage");
const result = await client.chunkPut(rawData);
console.log(`Chunk stored at: ${result.address}`);
console.log(`Cost: ${result.cost} atto tokens`);

// Retrieve the chunk
const retrieved = await client.chunkGet(result.address);
console.log(`Retrieved ${retrieved.length} bytes`);

if (!retrieved.equals(rawData)) throw new Error("Chunk round-trip mismatch!");
console.log("Chunk round-trip OK!");
